blog/content/post/2024-11-26-measured-boot-improvements.md: Add post

miczyg1 opened the pull request.

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+
+- [SeaBIOS](https://www.seabios.org/SeaBIOS) - for legacy BIOS compatibility
+- Linux payload/LinuxBoot/heads - a Linux kernel with optional initrd launched
+ directly by coreboot
```
Link to Heads?

miczyg1 replied:
ade8a4d823b46aa5642a83b03be7b36a62b2c808

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+### Solution
+
+We already have many pieces prepared earlier, like unified TPM support and a
+TCG-compliant event log. The work required to pass the vent log from coreboot
```
Spellchecker

miczyg1 replied:
I put it through grammarly, but vent is not a bad spelled word. It just doesn't fit the context :)

Will change to "event" of course.

miczyg1 replied:
ff3b5715948bb278461894321745458410058a69

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+
+## Summary
+
+TPM event log integrity is an excellent achievement for our Dasharo team. While we
```
```suggestion
TPM event log integrity is an excellent achievement for our Dasharo Team. While we
```

miczyg1 replied:
https://github.com/3mdeb/news-and-ideas/pull/613/commits/cff1006b43a13b63a542f10eed3a227911bce453#diff-053de20f23ef9f4af660d4f062e10dc313dc72f5ede66ec1a080c9c82ed815e6R294

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+      code. It may not be a big deal for modern platforms with big caches, but
+      rather for the old ones.
+    - coreboot supports only one algorithm in a single event log entry.
+      Support for an agile format would have to be implemented.
```
What would be the current mitigation for that?

Can't we just set up an admin password so a change of PCR banks would not be possible for someone who doesn't know the password?

miczyg1 replied:
In general even if an admin changes PCR banks, the change won't be applied or it will break measured boot.

The mitigation is like I described: be agile and calculate hashes for all currently enabled banks. Of course coreboot changes to event logging must follow.

pietrushnic replied:
> In general even if an admin changes PCR banks, the change won't be applied or it will break measured boot.

So why are we not saying what mitigates that? Dear admin, do not change banks because it will break the system's trustworthiness. The admin must comply with that policy and should be informed about it. If the admin doesn't want users to change that policy, even accidentally, then prevent that by protecting the option with a password. That is the current mitigation. There should be a better one with the correct implementation, but that still needs to be implemented. We may be working on that. Or could you have a reference point to follow up?

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+reproducible PCR2
+value](https://github.com/Dasharo/dasharo-issues/issues/455).
+
+## Summary
```
I think most of these sections are "Future Challanges" or something like that. The summary should stay consistent and mainly cover the call to action.

miczyg1 replied:
cff1006b43a13b63a542f10eed3a227911bce453

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+
+Improving the security of the firmware is a process. What is important is to
+take small steps and constantly move forward by improving things, one after
+another. So, let's collaborate and work together on the safety of our systems.
```
Safety and security are not the same thing, and in the case of measurements, it is more about trustworthiness because something may be trusted but not secure (e.g., we may expect the frame system to behave consistently in an unsecured manner like a honeypot). Trusted systems give us the most freedom because, based on attestation, we can decide if a platform is trustworthy for a given application.

miczyg1 replied:
b1d7e2548b7260f8f3587a8ae1a42b90203d6890

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+   Guard-created measurements must also be added to keep the PCR values
+   consistent with the TPM event log.
+
+Improving the security of the firmware is a process. What is important is to
```
This is summary IMHO

miczyg1 replied:
cff1006b43a13b63a542f10eed3a227911bce453

pietrushnic commented on blog/content/post/2024-11-26-measured-boot-improvements.md:
```diff
@@ -0,0 +1,222 @@
+hardware. And if you want to stay up-to-date on all things firmware security
+and optimization, be sure to [sign up for our
+newsletter](https://newsletter.3mdeb.com/subscription/PW6XnCeK6). Don't let
+your hardware hold you back, work with 3mdeb to achieve more!
```
I would appreciate adding in between the lines pitch about training related to security and trustworthiness:
https://3mdeb.com/training/
As well as a training pitch about the upcoming HWIO25 USA: https://hardwear.io/usa-2025/training/mastering-uefi-secure-boot.php

miczyg1 replied:
fac86ce1750d8cbe5b12e66b34e7208a234b786e

pietrushnic reviewed (changes requested):
Great post, thank you, my comments are not blocking.

miczyg1 commented:
@pietrushnic addressed all of the comments except https://github.com/3mdeb/news-and-ideas/pull/613#discussion_r1858880579 (treating it more like an open question)

macpijan reviewed (approved):
I reviewed as well, and resolved threads from @pietrushnic
